File: Insertion_Sort.java
import java.util.*;
public class Insertion_Sort
{
    int a[];
    int n;
    Insertion_Sort(int n)
    {
        this.n=n;
        a=new int[n];
    }
    void input()
    {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter array");
        for(int i=0;i<n;i++)
            a[i]=sc.nextInt();
    }
    void sort()
    {
        int h;
        int p;
        for(int i=0;i<n-1;i++)
        {
            h=a[i];
            p=i;
            for(int j=i;j<n-1;j++)
            {
                if(a[i]<a[j])
                {
                    h=a[j];
                    p=j;
                }
            }
            for(int k=i;k<p;k++)
                a[k]=a[k+1];
            a[p]=h;
        }
    }
    void display()
    {
        System.out.println(Arrays.toString(a));
    }
    public static void main(String args[])
    {
       Insertion_Sort obj=new Insertion_Sort(5);
        obj.input();
        obj.display();
        obj.sort();
        obj.display();
    }
}

File: Insertion_Sort.py
def insertionSort(alist):
   for i in range(0,len(alist),1):#i=0,1,2,3
     x = alist[i] #x=54, 26 ,55
     j = i # j=0,1 ,3   
     while j>0 and alist[j-1]>x: #j>0 54>26   93>54
         alist[j]=alist[j-1] # a[1]=54   a[3]=93
         j = j-1 # j=0 2
     alist[j]=x  #a[0]=26   a[2]=55        26,54,55,94
alist = [54,26,93,55,77,55,44,55,20]
insertionSort(alist)
print(alist)